src/urbit/app/chess.hoon

```diff
@@ -114,158 +156,241 @@
+                  %agent  [who.action %chess]
+                  %watch  /game/(scot %da game-id)/moves
               ==
+              ::
+              ::  add our new game to the list of active games
+              :*  %give  %fact  ~[/active-games]
+                  %chess-game  !>(new-game)
+              ==
+            ==
           %=  this
+            ::  remove our challenger from challenges-received
             challenges-received  (~(del by challenges-received) who.action)
-            challenges-sent  ?:  =(who.action our.bowl)
-                               (~(del by challenges-sent) our.bowl)
-                             challenges-sent
-            games  (~(put by games) game-id [new-game *chess-position | | |])
+            ::  if the poke came from our ship, delete our ship
+            challenges-sent
+              ?:  =(who.action our.bowl)
+                (~(del by challenges-sent) our.bowl)
+              challenges-sent
+            ::  put our new game into the map of games
+            games                (~(put by games) game-id [new-game *chess-position | | |])
           ==
         %decline-game
           =/  challenge  (~(get by challenges-received) who.action)
+          ::
+          ::  if challenge is null...
           ?~  challenge
+            ::  error!
+            ::  TODO: should display to frontend
             :_  this
             =/  err
               "no challenge to decline from {<who.action>}"
             :~  [%give %poke-ack `~[leaf+err]]
             ==
-          :-  :~  :*  %pass  /poke/challenge  %agent  [who.action %chess]
-                      %poke  %chess-decline-challenge  !>(~)
-                  ==
+          ::  if not, return [(list card) _this]
+          :-
+            :~
+              ::  tell our challenger we decline
+              :*  %pass  /poke/challenge  %agent  [who.action %chess]
+                  %poke  %chess-decline-challenge  !>(~)
               ==
+            ==
           %=  this
+            ::  remove our challenger from challenges-received
             challenges-received  (~(del by challenges-received) who.action)
           ==
         %offer-draw
           =/  game  (~(get by games) game-id.action)
+          ::
+          ::  if game is null...
           ?~  game
+            ::  error!
+            ::  TODO: should display to frontend
             :_  this
             =/  err
               "no active game with id {<game-id.action>}"
             :~  [%give %poke-ack `~[leaf+err]]
             ==
-          :-  :~  :*  %give  %fact  ~[/game/(scot %da game-id.action)/moves]
-                      %chess-draw-offer  !>(~)
-                  ==
+          ::  else, return [(list card) _this]
+          :-
+            :~
+              ::  for this move, we just offer a draw...
+              :*  %give  %fact  ~[/game/(scot %da game-id.action)/moves]
+                  %chess-draw-offer  !>(~)
```

Comment:
I actually prefer the existing style for a returned list of cards - once your eyes adjust to it, it's very easy to spot and it's clear that it's the logical end of a branch.